src/main/kotlin/com/coder/gateway/sdk/CoderRestClient.kt

```diff
@@ -182,7 +182,7 @@ open class CoderRestClient(
         // It is possible for there to be resources with duplicate names so we
         // need to use a set.
         return workspaces.flatMap { ws ->
-            resources(ws).filter { it.agents != null }.flatMap { it.agents!! }.map {
+            ws.latestBuild.resources.filter { it.agents != null }.flatMap { it.agents!! }.map {
```

Comment:
`ws.latestBuild.resources` can be empty if the workspace is off so I think we still need to fetch in that case, otherwise it will remove hosts from the SSH config.

Maybe something like this (untested):

```suggestion
            when (ws.latestBuild.status) {
                WorkspaceStatus.RUNNING -> ws.latestBuild.resources
                else -> client.resources(ws)
            }.filter { it.agents != null }.flatMap { it.agents!! }.map {
```

Reply:
Or something like `ws.latestBuild.resources.ifEmpty { resources(ws) }`

Reply:
done